src/view/xwayland.cpp

```diff
@@ -285,15 +305,17 @@ class wayfire_xwayland_view : public wayfire_xwayland_view_base
         if (xw->fullscreen)
             fullscreen_request(get_output(), true);
 
-        auto real_output = get_output()->get_layout_geometry();
-        if (!tiled_edges && !fullscreen && !parent)
+        wf::wlr_view_t::map(surface);
+
+        /* Freely-floating X windows might have set a geometry hint, try to
+         * take this into account */
+        if (!tiled_edges && !fullscreen && !parent && !self_positioned)
```

Comment:
I just realize this, but we may not need this at all. Can you try removing the if and its contents?

My logic is the following: if we had a configure request before mapping, we have already applied it so the view is at the correct place.

Reply:
This is needed or else xterm is placed as if it doesn't have decorations, without place enabled.

Reply:
As discussed on IRC, it doesn't really matter if place is disabled.